src/components/Tabs/Tabs.jsx

```
@@ -1 +1,35 @@
-export const Tabs = () => {};
+import React from 'react';
+
+export const Tabs = ({ tabs, activeTabId, onTabSelected }) => {
+  const activeTab = tabs.find(tab => tab.id === activeTabId) || tabs[0];
+
+  return (
+    <div data-cy="TabsComponent">
+      <div className="tabs is-boxed">
+        <ul>
+          {tabs.map(tab => (
+            // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-noninteractive-element-interactions
+            <li
+              key={tab.id}
+              data-cy="Tab"
+              className={tab.id === activeTab.id ? 'is-active' : ''}
```

This uses `activeTab.id` in the comparison which will throw if `activeTab` is `undefined` (empty `tabs`). This violates the requirement to "Handle empty tabs array gracefully (no errors). If empty, no tabs should render and TabContent should be empty or not rendered. onTabSelected should not be called in this case." Add a guard so this comparison only happens when `activeTab` is defined (or ensure `activeTab` is always an object when `tabs` is non-empty).